neo/rawio/neuralynxrawio/NcsBlocks.py

```diff
@@ -0,0 +1,416 @@
+        self.endTime = -1   # end time of last record, that is, the end time of the last
+                            # sampling period contained in the record
+
+    def __init__(self, sb, st, eb, et):
+        self.startBlock = sb
+        self.startTime = st
+        self.endBlock = eb
+        self.endTime = et
+
+    def before_time(self, rhb):
+        """
+        Determine if this block is completely before another block in time.
+        """
+        return self.endTime < rhb.startTime
+
+    def overlaps_time(self, rhb):
+        """
+        Determine if this block overlaps another in time.
+        """
+        return self.startTime <= rhb.endTime and self.endTime >= rhb.startTime
+
+    def after_time(self, rhb):
+        """
+        Determine if this block is completely after another block in time.
+        """
+        return self.startTime >= rhb.endTime
+
+
+class CscRecordHeader:
+    """
+    Information in header of each Ncs record, excluding sample values themselves.
+    """
+
+    def __init__(self, ncsMemMap, recn):
+        """
+        Construct a record header for a given record in a memory map for an NcsFile.
+        """
+        self.timestamp = ncsMemMap['timestamp'][recn]
+        self.channel_id = ncsMemMap['channel_id'][recn]
+        self.sample_rate = ncsMemMap['sample_rate'][recn]
+        self.nb_valid = ncsMemMap['nb_valid'][recn]
+
+
+class NcsBlocksFactory:
+    """
+    Class for factory methods which perform parsing of contiguous blocks of records
+    in Ncs files.
+
+    Model for times is that times are rounded to nearest microsecond. Times
+    from start of a sample until just before the next sample are included,
+    that is, closed lower bound and open upper bound on intervals. A
+    channel with no samples is empty and contains no time intervals.
+
+    Moved here since algorithm covering all 3 header styles and types used is
+    more complicated. Copied from Java code on Sept 7, 2020.
+    """
+
+    _maxGapLength = 5   # maximum gap between predicted and actual block timestamps still
+                        # considered within one NcsBlock
+
+    @staticmethod
+    def getFreqForMicrosPerSamp(micros):
+        """
+        Compute fractional sampling frequency, given microseconds per sample.
+        """
+        return 1e6 / micros
+
+    @staticmethod
+    def getMicrosPerSampForFreq(sampFr):
+        """
+        Calculate fractional microseconds per sample, given the sampling frequency (Hz).
+        """
+        return 1e6 / sampFr
+
+    @staticmethod
+    def calcSampleTime(sampFr, startTime, posn):
+        """
+        Calculate time rounded to microseconds for sample given frequency,
+        start time, and sample position.
+        """
+        return round(startTime + NcsBlocksFactory.getMicrosPerSampForFreq(sampFr) * posn)
+
+    @staticmethod
+    def _parseGivenActualFrequency(ncsMemMap, ncsBlocks, chanNum, reqFreq, blkOnePredTime):
```

Comment:
My understanding was that it was acceptable for private, non-public methods, to use this alternate convention, which frankly I prefer. But if you insist for consistency, I will change the private methods as well.

Reply:
I think it would be better to keep one style across the complete Neo codebase. I guess we might unintentionally already have too many exceptions of this, but if you don't mind I would prefer not to introduce more variability in naming schemes here.